llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -343,13 +421,32 @@ MachineInstr *InterBlockScheduling::latencyConverged(BlockState &BS) const {
   return nullptr;
 }
 
-bool InterBlockScheduling::updateFixPoint(BlockState &BS) {
+SchedulingStage InterBlockScheduling::updateFixPoint(BlockState &BS) {
+  if (BS.Kind != BlockType::Loop) {
+    return BS.FixPoint.Stage = SchedulingStage::SchedulingDone;
```

Comment:
I still find it confusing to return an assignment. The underlying value of the enum could change (and that will be an annoying thing to debug). How about 
```
BS.FixPoint.Stage = SchedulingStage::SchedulingDone;
return true;
```

Same for other cases when you assign and return at the same time.

Reply:
Clarified, I thought we were implicitly converting the enum to a `bool`, but the return type changed. All good.